### ISSUE TITLE
shortestPath() returns ArrayIndexOutOfBoundsException 

### ISSUE THREAD

**Issue by Grmpf101** [orientechnologies/orientdb#8013]
## Actual behavior  
After upgrading OrientDB from version 2.2.29 to 2.2.31 passing a direction to `shortestPath()` function results in the following error:
```
Error: java.lang.ArrayIndexOutOfBoundsException: 3
```

## Steps to reproduce  
I broke it down to this example:
```
CREATE CLASS Person EXTENDS V
CREATE CLASS Restaurant EXTENDS V
CREATE VERTEX Person SET name='Luca'
CREATE VERTEX Restaurant SET name='Dante', type='Pizza'
CREATE CLASS Eat EXTENDS E
CREATE EDGE Eat FROM ( SELECT FROM Person WHERE name='Luca' ) TO ( SELECT FROM Restaurant WHERE name='Dante' )
```

Performing shortestPath with direction `OUT`:
```
SELECT shortestPath($person[0].rid, $restaurant[0].rid, "OUT") LET $person = (select @rid from Person where name = "Luca"), $restaurant = (select @rid from Restaurant where name = "Dante")

Error: java.lang.ArrayIndexOutOfBoundsException: 3
```

Removing the direction works as expected:
```
SELECT shortestPath($person[0].rid, $restaurant[0].rid) LET $person = (select @rid from Person where name = "Luca"), $restaurant = (select @rid from Restaurant where name = "Dante")
+----+--------------+
|#   |shortestPath  |
+----+--------------+
|0   |[#73:0,#105:0]|
+----+--------------+

1 item(s) found. Query executed in 0.007 sec(s).
```

**Comment by luigidellaquila** [comment]
Hi @Grmpf101 

Thank you very much for reporting, it seems to be a regression, I'm checking it now

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @Grmpf101 

I just pushed a fix for this, it will be released with 2.2.32

Thanks

Luigi
